### DISCUSSION TITLE
Visualization

### DISCUSSION THREAD

**Discussion by meenow708** [iTwin/community#22] [General]
Hi,

I have this code which is not going well, So i done with the coding there is not error found but the the coding is not reflecting in the **Itwin view** can anyone help out!

```
import { QueryRowFormat } from "@itwin/core-common";
import { IModelConnection, ScreenViewport } from "@itwin/core-frontend";

export class Visualization {

  public static hideHouseExterior = async (vp: ScreenViewport) => {

    const categoryIds = await Visualization.getCategoryIds(vp.iModel);
    vp.changeCategoryDisplay(categoryIds, false);
    
    console.log(categoryIds);

  }

  private static getCategoryIds = async (iModel: IModelConnection) => {

    const categoriesToHide = ["'RevitDynamic.SWall'"];

    const query = `SELECT ECInstanceId FROM RevitDynamic.SWall(${categoriesToHide.toString()})`;


    const result = iModel.query(query, undefined, { rowFormat: QueryRowFormat.UseJsPropertyNames });
    const categoryIds = [];

    for await (const row of result)
        categoryIds.push(row.id);


    return categoryIds;
  }
}
```

**App.tsx file code:**

![image](https://github.com/iTwin/community/assets/157709671/e5d69e79-47ea-49c1-b91f-6129aeedac90)

![Screenshot 2024-02-01 165724](https://github.com/iTwin/community/assets/157709671/15c9092a-3611-4d43-a17e-218ffc8f1718)

**Comment by pmconne** [comment]
> the coding is not reflecting in the Itwin view 

What does this mean?

**Reply by meenow708** [reply]
See below attached image:

![image](https://github.com/iTwin/community/assets/157709671/587d8b37-c359-4e61-b914-32943c2f25ff)

i wrote this visualization code to hide my wall-category from the view but it is not hiding from the view.

**Comment by pmconne** [comment]
Use the debugger to confirm `hideHouseExterior` is being called, that your query is returning expected results, etc.

**Comment by meenow708** [comment]
i dint understood can you please have some examples for better understanding

**Comment by pmconne** [comment]
Use ctrl-shift-I to open your browser's development tools.
Stick a breakpoint on the last line of `getCategoryIds`.
Run your app.
Confirm your breakpoint gets hit. If it does, inspect the contents of `categoryIds` to confirm it contains the categories you expect it to.

**Reply by Zubair-KPM** [reply]
thanks
